packages/custom-storage/src/sql/SQLStorage.ts

```diff
@@ -532,16 +534,23 @@ export class SQLStorage {
   /**
    * Checks if all the tables for SQLStorage exist
    */
-  private async _checkMissingTables() {
-    const requiredTables = [this._tableName, this._datasetsTableName, this._datasetsVisTableName];
-
-    const checksTablesAreReady = requiredTables.map((table) => this._sql.query(`SELECT to_regclass('${table}')`));
-    const results = await Promise.all(checksTablesAreReady);
-    const missingTables = results.some((response: any) => {
-      const tableIsMissing = (response.rows[0].to_regclass === null);
-      return tableIsMissing;
+  private _checkMissingTables() {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const requiredTables = [this._tableName, this._datasetsTableName, this._datasetsVisTableName];
+
+        const checksTablesAreReady = requiredTables.map((table) => this._sql.query(`SELECT to_regclass('${table}')`));
+        const results = await Promise.all(checksTablesAreReady);
+        const missingTables = results.some((response: any) => {
+          const tableIsMissing = (response.rows[0].to_regclass === null);
+          return tableIsMissing;
+        });
+
+        resolve(missingTables);
+      } catch (err) {
+        reject(err);
```

Comment:
Out of curiosity, what kind of errors have you found here?

Reply:
No errors yet, but as long as I have to create a Promise, I always prefer to make use of the reject function for possible errors